### PR TITLE
Add a service to get members from discord

### PR DESCRIPTION
**Pull request by RitikJaiswal75**
I'm excited to inform you that I've developed a new method to obtain details regarding the individuals in our Discord community. This feature has proven incredibly useful in monitoring activity levels and identifying those less engaged. If there's anything specific you'd like me to investigate, please don't hesitate to ask!

We are not writing tests for it as it is just a fetch call and fetch calls are already tested.

we are not throwing an error in service as it will be present in the file we will consume the service.

### CODE REVIEW

File: services/discordService.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    const response = await fetch(`${DISCORD_BASE_URL}/discord-members`, {
+      method: "GET",
+      headers: { "Content-Type": "application/json", Authorization: `Bearer ${authToken}` },
+    })
+      .then((response, err) => {
+        return response.json();
+      })
+      .then((data) => {
+        return data;
+      });
```
Comment (heyrandhir) [inline services/discordService.js:21]: can you please confirm if this `.then` is needed. as it is just returning the data which it is receiving. Removing this should still make the code work.

File: services/discordService.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    return response;
+  } catch (err) {
+    return [{ message: "Oops an internal error occured" }];
```
Comment (heyrandhir) [inline services/discordService.js:24]: why are we returning error message in a array 🤔

**Review by heyrandhir — CHANGES_REQUESTED** [review]

File: services/discordService.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+  });
+
+  try {
+    const response = await fetch(`${DISCORD_BASE_URL}/discord-members`, {
+      method: "GET",
+      headers: { "Content-Type": "application/json", Authorization: `Bearer ${authToken}` },
+    })
+      .then((response, err) => {
+        return response.json();
+      })
```
Comment (heyrandhir) [inline services/discordService.js:18]: Nit : we can use async/await instead of promise chaining

**Review by heyrandhir — CHANGES_REQUESTED** [review]

File: services/discordService.js (hunk 1 above)
Comment (RitikJaiswal75) [reply services/discordService.js:21]: yes i tried it normally but id did not work as our cfw sends nested response, which wasn't accessible normally.

Also, could you provide reference where it sayswe should not use `.then`?

File: services/discordService.js (hunk 2 above)
Comment (RitikJaiswal75) [reply services/discordService.js:24]: to maintain consistency in data transfer object

File: services/discordService.js (hunk 3 above)
Comment (RitikJaiswal75) [reply services/discordService.js:18]: tried, but wasn't working

File: services/discordService.js (hunk 1 above)
Comment (heyrandhir) [reply services/discordService.js:21]: Certainly! While there are no strict rules about using `async` and `await` in place of promise chaining,  using async await makes code easier to understand. If you see this mdn doc [here](https://developer.mozilla.org/en-US/docs/Learn/JavaScript/Asynchronous/Promises) it says 
```note
The async and await keywords make it easier to build an operation from a series of consecutive asynchronous function calls, avoiding the need to create explicit promise chains, and allowing you to write code that looks just like synchronous code.
```

**Review by heyrandhir — APPROVED** [review]

File: services/discordService.js [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+const DISCORD_BASE_URL = config.get("services.discordBot.baseUrl");
+
+const getDiscordMembers = async () => {
+  const authToken = jwt.sign({}, config.get("rdsServerlessBot.rdsServerLessPrivateKey"), {
+    algorithm: "RS256",
+    expiresIn: config.get("rdsServerlessBot.ttl"),
+  });
+
+  try {
+    const response = await fetch(`${DISCORD_BASE_URL}/discord-members`, {
```
Comment (DashDeipayan) [inline services/discordService.js:12]: Why was await and then used together, use either await(preferred) or then

File: services/discordService.js (hunk 2 above)
Comment (DashDeipayan) [inline services/discordService.js:24]: We must log the error

**Pull request merged**